packages/openapi2-parser/lib/schema.js

```diff
@@ -271,6 +266,15 @@ class DataStructureGenerator {
       element = new typeGeneratorMap[type]();
     } else if (type) {
       throw new Error(`Unhandled schema type '${type}'`);
+    } else {
+      element = new this.minim.elements.Enum();
+      element.enumerations = [
+        new this.minim.elements.String(),
+        new this.minim.elements.Number(),
+        new this.minim.elements.Boolean(),
+        this.generateArray(schema),
+        this.generateObject(schema),
+      ];
```

Comment:
And here we create enum with all datatypes. Also in case schema has for eg. items it will create array with items of given type.

Reply:
Exactly, we don't support other validation keywords for the primitive types so we can create an empty primitive element (boolean, string, number) in those cases.

In the cases of array and object we support more keywords so there's handling of using those existing functions which process the items/properties/required and similiar keywords.